pkg/autodiscovery/README.md

```diff
@@ -1,15 +1,92 @@
 # package `autodiscovery`
 
-This package is a core piece of the agent. It is responsible for collecting check configurations from different sources (see package [config providers](https://github.com/DataDog/datadog-agent/tree/main/pkg/autodiscovery/providers)) and then schedule or unschedule integration configurations with the help of the schedulers.
+This package manages configuration for dynamic entities like pods and containers.
 
-It is also responsible for listening to container-related events and resolve template configurations that would match them.
+## Architecture
 
-## `AutoConfig`
+The high-level architecture of this package is implemented by the AutoConfig type, and looks like this:
 
-As a central component, `AutoConfig` owns and orchestrates several key modules:
+```
+Kubernetes
+    API──────┐
+             │
+Cluster      │ ┌────────────────┐
+  Agent────┐ │ │ Workload Meta  │
+           │ │ └──┬──────────┬──┘
+ Static    │ │    │          │
+  Files──┐ │ │    │          │
+         │ │ │    │          │
+    ┌────▼─▼─▼────▼──┐    ┌──▼─────────────┐
+    │Config Providers│    │   Listeners    │
+    └──┬─────┬───────┘    └────────┬───────┘
+       │     │                     │
+       │     │                     │
+       │     │templates    services│
+       │     └────────► │ ◄────────┤
+       │                │          │
+       │            reconcile      │
+       │                │          │
+       │non-template    │          │
+       │configs         │          │
+       │                │          │
+       │                ▼          │
+       │         ┌─────────────┐   │
+       └────────►│metascheduler│◄──┘
+                 └─────────────┘
+```
```

Comment:
Question: Are there services that go directly from the listener to the metascheduler without being reconciled with a template?

Reply:
Just to add context: this a technical debt to make the logs agent learn about all containers and support `logs_config.container_collect_all`. It should be replaced by subscribing directly to workloadmeta in the coming releases 🤞

Reply:
That's what I'm working on -- but aspirational docs are an anti-pattern, so let's doc it as it is.

Reply:
Of course, we should document it as it is 👌